src/utility.h

```diff
@@ -583,9 +583,13 @@ class Guard
 class QDLLEXPORT TimeThis
 {
 public:
-  TimeThis(QString what={});
+  TimeThis(const QString& what={});
   ~TimeThis();
 
+  TimeThis(const TimeThis&) = delete;
+  TimeThis& operator=(const TimeThis&) = delete;
+
+  void start(const QString& what={});

```

Comment:
Nice addition, maybe it would be worth it to add some comments since it evolved into something a little more complex now.